hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/protocol/SCMSecretKeyProtocolOm.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.hdds.protocol;
+
+import org.apache.hadoop.ipc.ProtocolInfo;
+import org.apache.hadoop.security.KerberosInfo;
+
+import static org.apache.hadoop.hdds.scm.ScmConfig.ConfigStrings.HDDS_SCM_KERBEROS_PRINCIPAL_KEY;
+
+/**
+ * The client protocol to access secret key from OM.
+ */
+@ProtocolInfo(protocolName =
+    "org.apache.hadoop.hdds.protocol.SCMSecretKeyProtocolOm",
+    protocolVersion = 1)
+@KerberosInfo(
+    serverPrincipal = HDDS_SCM_KERBEROS_PRINCIPAL_KEY,
+    // TODO: move OMConfigKeys.OZONE_OM_KERBEROS_PRINCIPAL_KEY to hdds-common.
```

Comment:
Why this todo?

Reply:
This config key is used in a lot of places (more than 25 usages), I want to avoid including those changes in this PR. Filed a ticket: HDDS-8540